src/runtime/recipe/handle-connection.ts

```diff
@@ -77,15 +76,22 @@ export class HandleConnection {
   getQualifiedName() { return `${this.particle.name}::${this.name}`; }
   get tags() { return this._tags; }
   get type() {
-    if (this._type) {
-      return this._type;
+    if (this._rawType) {
```

Comment:
a type at this level is now a resolved type. Rename _rawType to _resolvedType. Also remove the underscore and make it private :)

Reply:
done - renamed and made private.